### PR TITLE
Update packages and solidity contract functions

### LINKED ISSUE #309: [TASK] Use v0.5.0 solidity contract files and update the relayer tx-relay transact functions
# Overview
We've updated the protocol-solidity infra to version 0.5.0. We need to update the `webb-rs` and after doing so, update the `relayer` to use these contracts. Most notably, the tx-relay for EVM will fail. This needs to be updated with the new structure for function args.

## Checklist
- [ ] Update the webb-rs version once the contract fixtures are updated.
- [ ] Update the `relayer` `tx-relay` package for EVM.
- [ ] Update the tests to ensure they pass.

### PR DESCRIPTION
**Pull request by salman01zp**
## Summary of changes
- [x] update webb-tools package to` v0.1.4-117`
- [x] update solidity contract package to `v0.5.3`
- [x] update webb-rs to `v0.5.11`
- [x] update transact function for evm
- [x] update Event watcher to support latest solidity contracts

### Reference issue to close (if applicable)
- Closes #309 

-----
### Code Checklist 

- [x] Tested
- [ ] Documented

### CODE REVIEW

**Comment by shekohex** [comment]
Blocked on https://github.com/webb-tools/protocol-solidity/pull/220

**Comment by shekohex** [comment]
This now merged: https://github.com/webb-tools/protocol-solidity/pull/220
So we can update the packages and it should work and compile again.

**Comment by drewstone** [comment]
Yep we should update to 0.5.1

File: event-watchers/evm/src/open_vanchor/open_vanchor_deposit_handler.rs [hunk 1]
```diff
@@ -80,25 +80,7 @@ where
...
-
-        if event_data.leaf_index.as_u32() % 2 == 0 {
-            tracing::debug!(
-                leaf_index = %event_data.leaf_index,
-                is_even_index = %event_data.leaf_index.as_u32() % 2 == 0,
-                "Open VAnchor new leaf index does not satisfy the condition, skipping proposal.",
-            );
-            return Ok(());
-        }
-
```
[the remark this answers is not in the document]
Comment (salman01zp) [reply event-watchers/evm/src/open_vanchor/open_vanchor_deposit_handler.rs:101]: OpenVanchor contract has been updated  [here](https://github.com/webb-tools/protocol-solidity/blob/main/packages/contracts/contracts/vanchors/extensions/OpenVAnchor.sol#L153-#L156) to insert only one commitment after deposit.

[the remark this answers is not in the document]
Comment (drewstone) [reply event-watchers/evm/src/open_vanchor/open_vanchor_deposit_handler.rs]: Yep should only be 1 insertion at a time since they're all public.

**Comment by drewstone** [comment]
Great work @salman01zp !

**Pull request merged**